atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraService.java

```diff
@@ -180,7 +182,7 @@ public Set<CassandraServer> refreshTokenRangesAndGetServers() {
             tokenMap = tokensInterner.intern(newTokenRing.build());
             logHostToDatacenterMapping(hostToDatacentersThisRefresh);
             hostToDatacenter = hostToDatacentersThisRefresh;
-            return servers;
+            return ImmutableSet.copyOf(servers);
```

Comment:
could we make servers an immutable set builder instead? I assume that'd be no worse